### PR TITLE
Refactor styles header

### PR DESCRIPTION
**Pull request by LesiaM-T**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/ita-social-projects/EventsExpress-Frontend/projects/1)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of change

Refactored header after demo. Created separate rightBlock and  dropdownMenu components, changed  classNames

## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/components/Header/headerRightBlock/headerRightBlock.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          </ToggleButton>
+        </div>
+        <FaSearch className="header-icons__item" />
+        <FaStar className="header-icons__item" />
+      </div>
+      <div>
+        {!id && (
+          <AuthComponent onlyAnonymous>
+            <ModalWind
+              renderButton={action => (
```
Comment (VadymKharyton) [inline src/components/Header/headerRightBlock/headerRightBlock.js]: This function doesn't depend on any parameter or local variable. Can it be extracted and moved outside of component?

File: src/components/Header/headerRightBlock/headerRightBlock.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+const { SIGN_IN } = headerConstants;
+
+const HeaderRightBlock = user => {
+  const { id } = user.id ? user : {};
+
+  return (
+    <div className="header__right-block">
+      <div className="header-icons">
+        <div className="language-toggle">
```
Comment (VadymKharyton) [inline src/components/Header/headerRightBlock/headerRightBlock.js]: This div is redundant. You can pass `className` prop directly to the `ToggleButton` component

File: src/components/shared/toggleButton/toggleButton.js [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
-    ></div>
+    >
+      {children}
+    </div>
   );
 };
-export default TogleButton;
+
+ToggleButton.defaultProps = {
+  children: React.createElement("div"),
```
Comment (VadymKharyton) [inline src/components/shared/toggleButton/toggleButton.js]: Replace with empty fragment to not create extra `div` tag

File: src/components/Header/headerRightBlock/headerRightBlock.js (hunk 2 above)
Comment (LesiaM-T) [reply src/components/Header/headerRightBlock/headerRightBlock.js]: yes, thank you

File: src/components/Header/headerRightBlock/headerRightBlock.js (hunk 1 above)
Comment (LesiaM-T) [reply src/components/Header/headerRightBlock/headerRightBlock.js]: Done

File: src/components/shared/toggleButton/toggleButton.js (hunk 3 above)
Comment (LesiaM-T) [reply src/components/shared/toggleButton/toggleButton.js]: Done

File: src/components/shared/toggleButton/toggleButton.js [hunk 4]
```diff
@@ -1,15 +1,26 @@
 import React from "react";
+import PropTypes from "prop-types";
 import "./toggleButton.scss";
 
-const TogleButton = () => {
+const ToggleButton = ({ children }) => {
```
Comment (VadymKharyton) [inline src/components/shared/toggleButton/toggleButton.js:5]: Does it works as expected? You made right changes, where you passed className into this component(in HeaderRightBlock), but this prop didn't handle inside the component

Comment (LesiaM-T) [reply src/components/shared/toggleButton/toggleButton.js:5]: It was not working, now I see I should handle it in props. But in the end, after refactoring, this class changes only one small margine, thats why I had not seen its not workin. I will drop not nessesary class and pass the margine to ToggelButton. Thank you.

**Review by VadymKharyton — APPROVED** [review]

**Pull request merged**